### ISSUE TITLE
ZXCaptureDelegate Sometimes repeated execution

### ISSUE THREAD

**Issue by duxiaoyan** [TheLevelUp/ZXingObjC#328]
- (void)captureResult:(ZXCapture *)capture result:(ZXResult *)result
{
     NSLog(@"%@",result.text);    
    [self.capture stop];
}

 Sometimes repeated execution，printed many nslog

**Comment by miguelwicht** [comment]
I have the same problem.

**Comment by niaj-twinbit** [comment]
Same problem here... is there anyone who solved it?

**Comment by kientux** [comment]
It said to be fixed in this #200 , but I'm still having the same problem. He put a check in the code:
```
- (void)captureOutput:(AVCaptureOutput *)captureOutput
didOutputSampleBuffer:(CMSampleBufferRef)sampleBuffer
       fromConnection:(AVCaptureConnection *)connection {
    if (!self.running) return;
    ....
}
```
but it seems that at that point, the capture is actually still running so the check is pointless.

My fix is that I check for `capture.running` in delegate method:

```
- (void)captureResult:(ZXCapture *)capture result:(ZXResult *)result {
    if (!result) return;
    if (!capture.running) return;
    ....
}
```

and it worked. So maybe he should check for `capture.running` right before calling delegate method.
